### PR TITLE
Fix Windows external CI version template to use OOT

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
**What type of PR is this?**

/kind failing-test

**What this PR does / why we need it**: The external cloud provider CI version template has a bug as the windows deployment is not using external cloud-provider so it's actually testing in-tree for windows.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Review by CecileRobertMichon — COMMENTED** [review]
/assign @marosset @lzhecheng

File: templates/test/ci/cluster-template-prow-external-cloud-provider-ci-version.yaml [hunk 1]
```diff
@@ -523,8 +523,8 @@ spec:
           kubeletExtraArgs:
             azure-container-registry-config: c:/k/azure.json
             cloud-config: c:/k/azure.json
-            cloud-provider: azure
-            feature-gates: WindowsHostProcessContainers=true
```
Comment (CecileRobertMichon) [inline templates/test/ci/cluster-template-prow-external-cloud-provider-ci-version.yaml:527]: /hold

this shouldn't be gone

Comment (CecileRobertMichon) [reply templates/test/ci/cluster-template-prow-external-cloud-provider-ci-version.yaml:527]: /hold cancel

**Comment by lzhecheng** [comment]
/lgtm
Thank you!

**Review by marosset — APPROVED** [review]

**Pull request merged**